feat: create campaign main page

Esinko opened the pull request:
- Include to-be-donated devices' technical information and images
- Include placeholders for applying for a donation
- Include some basic campaign information

Esinko committed:
feat: create campaign main page

- Include to-be-donated devices' technical information and images
- Include placeholders for applying for a donation
- Include some basic campaign information

Esinko committed:
fix: fix getServerSideProps function and import

- I forgor

Eldemarkki commented on pages/_app.tsx:
```diff
@@ -13,6 +13,7 @@ const pages = [
   { label: "Projektit", path: "/projects" },
   { label: "Tietoa meistä", path: "/about" },
   //{ label: "Tietoa", path: "/about-us" }
+  { label: "Koneet kiertoon", path: "/computerSharing" }
```
Personal preference, but I think `/computer-sharing` would be better.

Esinko replied:
Agreed.

Eldemarkki commented on pages/computerSharing.tsx:
```diff
@@ -0,0 +1,314 @@
+      </Head>
+      <Content>
+        <H1>Koneet kiertoon kampanja</H1>
+        <p><b>Tämän sivun tietoja täydennetään vielä. Kampanjan alkamisesta tiedoitetaan erikseen.</b></p>
```
tiedoitetaan -> tiedotetaan

Esinko replied:
Will fix.

Eldemarkki commented on pages/computerSharing.tsx:
```diff
@@ -0,0 +1,314 @@
+        <p><b>Tämän sivun tietoja täydennetään vielä. Kampanjan alkamisesta tiedoitetaan erikseen.</b></p>
+        <p>
+          Testausserveri ry organisoi yksityisen lahjoittajan lahjoittamien tietokoneiden ja palvelimien jakamisen yhdistyksen jäsenille ja hyväntekeväisyyteen.
+          Kampanjan tavoitteena on antaa turhiksi jääneille vanhoille, mutta edelleen moneen eri käyttötarkoituksiin sopiville, tietokoneille ja palvelimille uusi elämä.
```
moneen eri käyttötarkoituksiin -> moniin eri käyttötarkoituksiin

Eldemarkki replied:
sopiville, tietokoneille -> sopiville tietokoneille

Esinko replied:
Will fix.

Eldemarkki commented on pages/computerSharing.tsx:
```diff
@@ -0,0 +1,314 @@
+                "/computer-sharing-assets/worksationBackSide.png"
+              ].map((str) => ({ url: str }))
+            }/> 
+            <p>Huom. kuvassa olevasta koneesta puuttuu kiintolevy. Tämä oli kuvaajan moka. Kaikissa koneissa on joku ylhäällä mainituista kiintolevyistä.</p>
```
joku -> jokin

Esinko replied:
Will fix.

Eldemarkki commented on pages/computerSharing.tsx:
```diff
@@ -0,0 +1,314 @@
+          <Device>
+            <h3>HP EliteBook 840 G5 (? kpl)</h3>
+            <p>
+              Nämä päivittäisessä toimistokäytössä olleet kannettavat ovat hyvässä kunnossa muutamia pintanaarmuja lukuunottamatta.
```
lukuunottamatta -> lukuun ottamatta

Eldemarkki commented on pages/computerSharing.tsx:
```diff
@@ -0,0 +1,314 @@
+        <Section>
+          <h2>Lahjoituksen saamisen kynnysehdot</h2>
+          <p>
+            Koneet kiertoon -kampanjan tavoitteena on yhdistyksemme tavoitteiden mukaisesti edistää tietotekniikka- ja kyberturvallisuus harrastuneisuutta.
```
kyberturvallisuus harrastuneisuutta -> kyberturvallisuusharrastuneisuutta

Eldemarkki commented on pages/computerSharing.tsx:
```diff
@@ -0,0 +1,314 @@
+          <h2>Lahjoituksen saamisen kynnysehdot</h2>
+          <p>
+            Koneet kiertoon -kampanjan tavoitteena on yhdistyksemme tavoitteiden mukaisesti edistää tietotekniikka- ja kyberturvallisuus harrastuneisuutta.
+            Lahjoutuksen vastaanottaminen edellyttää kyselyyn vastaamista, jossa toivomme lahjoitusta hakevien kirjoittavan hieman laajemmin eri käyttötarkoituksista, joihin lahjoitettava laite/laitteet menisivät.
```
Lahjoutuksen -> Lahjoituksen

Eldemarkki commented on pages/computerSharing.tsx:
```diff
@@ -0,0 +1,314 @@
+            TBD
+          </p>
+          <p>
+            Vastaa kyselyyn ja hae lahjoutusta:
```
lahjoutusta -> lahjoitusta

Eldemarkki commented on pages/computerSharing.tsx:
```diff
@@ -0,0 +1,314 @@
+              [
+                "/computer-sharing-assets/workstationFrontSide.png",
+                "/computer-sharing-assets/workstationInternalsSide.png",
+                "/computer-sharing-assets/worksationBackSide.png"
```
Typo in filename (worksationBackSide -> workstationBackSide) but probably doens't need fixing as it's not displayed to the user

ahnl reviewed (approved):
[Hyvältä näyttää niinku kuuluuki!](https://www.youtube.com/watch?v=pFCJBhj6ooc)

ahnl reviewed (dismissed):
Taino, Eldermarkin mainitsemat typot ja responsiivisuus kondikseen.

<img width="393" alt="image" src="https://github.com/Testausserveri/testausserveri.fi/assets/46541386/d2644f70-e2de-4d94-8437-7d46a0307232">

Esinko committed:
feat: device element responsivity

- The device element now aligns content top-bottom when side-side is not possible within element bounds
- Includes some minor refactoring

Esinko committed:
fix: remove incorrect margin from gallery

- Removes extra margin from gallery image selection while under 776px (left margin)

Esinko committed:
fix: typos in content and filenames

- Changes campaign page name to computer-sharing.tsx
- Fixed typo in image asset name
- Content typos fixed

Esinko committed:
fix: add margin to small gallery images

ahnl committed:
fix: minor styles

ahnl committed:
fix: more minor styles

Chicken commented:
Pikkupräntti/ehdot vois tuoda sivun pohjalta ylös jossa ne on helposti luettavissa

ahnl commented:
Lisää keskustelua ja ideointia kampanjasta lauantaina 27.5. #yhdistys-kanavalla:
https://discord.com/channels/697710787636101202/1039504929468989440/1112095133291266098

ahnl committed:
Rename computer-sharing.tsx to koneet-kiertoon.tsx

Eldemarkki commented on pages/_app.tsx:
```diff
@@ -13,6 +13,7 @@ const pages = [
   { label: "Projektit", path: "/projects" },
   { label: "Tietoa meistä", path: "/about" },
   //{ label: "Tietoa", path: "/about-us" }
+  { label: "Koneet kiertoon", path: "/computer-sharing" }
```
Update this to `/koneet-kiertoon`, as the path was updated in https://github.com/Testausserveri/testausserveri.fi/pull/99/commits/0fe39eecc43410bb1f9ba8ed1ace1f8f226f4211

Esinko replied:
Fixed in [e4125c5](https://github.com/Testausserveri/testausserveri.fi/pull/99/commits/e4125c56cf8cab9d9ec0c55e15c9f1b8876d5c51)

Eldemarkki commented on pages/koneet-kiertoon.tsx:
```diff
@@ -0,0 +1,411 @@
+                <p><b>Piirisarja:</b> ?</p>
+                <p>
+                  <b>Näytönohjain:</b>
+                  <br />
```
This section (along with "kiintolevy", "laajennuspaikat" and "liitännät" sections) should maybe be an `<ul>`. It would be semantically more correct, and I think the `&nbsp;` looks a bit weird

Esinko replied:
This is true, but we'll fix this later.

Eldemarkki commented on pages/koneet-kiertoon.tsx:
```diff
@@ -0,0 +1,411 @@
+  margin-top: 3rem;
+  margin-bottom: 1rem;
+
+  h2 {
```
Personal opinion, but I don't like putting these child components inside styled-components' CSS. Instead, I create a new component for them: `const SectionTitle = styled.h2...`. It makes the code more maintainable and readable.

Esinko replied:
This change can be implemented later. Feel free to make a PR after this one is merged.

Eldemarkki commented on pages/koneet-kiertoon.tsx:
```diff
@@ -0,0 +1,411 @@
+  flex-direction: row;
+  flex-wrap: wrap;
+  background-color: rgba(108, 108, 108, 0.09);
+  padding: 1rem;
```
This `padding: 1rem;` should be removed because it's redefined on the line below.

Esinko replied:
Fixed in [e4125c5](https://github.com/Testausserveri/testausserveri.fi/pull/99/commits/e4125c56cf8cab9d9ec0c55e15c9f1b8876d5c51)

Eldemarkki commented on pages/koneet-kiertoon.tsx:
```diff
@@ -0,0 +1,411 @@
+  color: rgba(255,255,255,0.6);
+  width: auto !important;
+  text-align: left;
+  margin: 0 0 1rem 0 !important;
```
Could you just use `margin-bottom: 1rem !important;`?

Esinko replied:
Fixed in [e4125c5](https://github.com/Testausserveri/testausserveri.fi/pull/99/commits/e4125c56cf8cab9d9ec0c55e15c9f1b8876d5c51)

Eldemarkki commented on pages/koneet-kiertoon.tsx:
```diff
@@ -0,0 +1,411 @@
+      height: 25rem !important;
+
+      ul {
+        margin-top: 1rem;
```
The `margin-top: 1rem;` and `margin-bottom: 1rem;` can be combined into `margin-block: 1rem;`

Esinko replied:
Fixed in #[e4125c5](https://github.com/Testausserveri/testausserveri.fi/pull/99/commits/e4125c56cf8cab9d9ec0c55e15c9f1b8876d5c51)

Eldemarkki commented on pages/koneet-kiertoon.tsx:
```diff
@@ -0,0 +1,411 @@
+            <br />
+            <b>Kaikkien laitteiden yhteydessä annetaan kyseiseen laitteeseen sopiva virtalähde/kaapeli.</b>
+          </p>
+          <Device>
```
Probably not necessary (and maybe also a personal opinion), but I would personally refactor the 3 computer models into a reusable component. That way we could ensure consistent styling, reduce duplicated code and make the code cleaner overall.

So for example
```tsx
<ComputerModel
    description="Nämä HP:n Workstation koneet ovat..."
    name="HP Workstation Z240"
    amount={{
        remaining: 13,
        max: 15
    }}
    specs={{
        case: "Torni",
        cpu={{
            model: "i7-7700",
            cores: 4
            ...
        }}
        gpu={[
             ...,
             ...,
        ]}
    }}
    note="Huom. kuvassa olevasta koneesta..."
/>
```

But as I said, might be overkill in this case.

Esinko replied:
This is a great idea! Good future-proofing too. However, we'll deal with this later. Again, feel free to make a PR.

Esinko committed:
fix: proposed changes by @Eldemarkki and final touches

Esinko commented:
Will merge now. Further changes should be done through PRs towards the coal branch.
